### PR TITLE
performance optimizations

### PR DESCRIPTION
**Pull request by amirouche**

### CODE REVIEW

**Comment by amirouche** [comment]
Before:

```
setup_ahocorapy: 1.1680949940000573
search_ahocorapy: 2.1770270529996196
```

After:

```
setup_ahocorapy: 1.1381680919994324
search_ahocorapy: 1.6007716670001173
```

**Comment by FrederikP** [comment]
:+1: 
Thanks for this PR. On first sight, these changes make sense. I will verify performance.

In the meantime: Can you fix the PEP8 issue identified by the CI build?

```
__________________________________ PEP8-check __________________________________

/home/travis/build/abusix/ahocorapy/src/ahocorapy/keywordtree.py:120:58: E127 continuation line over-indented for visual indent

                                                         zero_state))

                                                         ^
```

File: src/ahocorapy/keywordtree.py [hunk 1]
```diff
@@ -112,13 +112,14 @@ def search_all(self, text):
...
             text = text.lower()
-        current_state = self._zero_state
+        zero_state = self._zero_state
+        current_state = zero_state
         for idx, symbol in enumerate(text):
             current_state = current_state.transitions.get(
-                symbol, self._zero_state.transitions.get(symbol,
-                                                         self._zero_state))
+                symbol, zero_state.transitions.get(symbol,
+                                                         zero_state))
```
Comment (FrederikP) [inline src/ahocorapy/keywordtree.py:120]: Please fix the formatting to be PEP8 compliant as checked by the CI.

**Review by FrederikP — COMMENTED** [review]
Performance looks fine, so as soon as you have fixed formatting, I will gladly merge this PR, update performance measurements in the README and release a new version to pypi. Thanks!

File: src/ahocorapy/keywordtree.py (hunk 1 above)
Comment (amirouche) [reply src/ahocorapy/keywordtree.py]: Done. Thanks in advance for the release :)

**Comment by FrederikP** [comment]
Released as version 1.6.1

The performance fix did not have any meaningful effect on the library run with pypy, but that was kind of expected. But it does have a positive effect when running the library with cpython. Thanks a lot!

**Comment by amirouche** [comment]
Did you tag the release?

**Comment by FrederikP** [comment]
> Did you tag the release?

Good point. Now I did. But it should've been available on pypi regardless of the tag.

**Pull request merged**